code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -230,6 +242,8 @@
 	var/reagents_per_slice = reagents.total_volume/slices_num //Figure out how much reagents each slice inherits (losing slices loses reagents)
 	for(var/i = 1 to (slices_num - slices_lost)) //Transfer those reagents
 		var/obj/slice = new slice_path(src.loc)
+		if(contents.len)
+			slice.contents += contents[1] //Our contents move to the slices
```

Comment:
Don't you know that contents is internally maintained?

Adding contents[1] to another items contents is the equivalent of setting the loc of contents[1] to the new object, which internally results in the removal of contents[1] from contents.